api/src/env.ts

```diff
@@ -145,7 +145,7 @@ function getEnv() {
 	}
 
 	// Default to env vars plain text files
-	return dotenv.parse(fs.readFileSync(configPath).toString());
+	return dotenv.parse(fs.readFileSync(configPath, { encoding: 'utf8' }));
```

Comment:
Lets await `fs.readFile` instead of using sync. Even though this is once on startup, we shouldn't block the thread

Reply:
Using await `fs.readFile` here would mean that `getEnv` has to be an async function and we would have to await `getEnv` in `let env...`, but this is not supported:
`Top-level 'await' expressions are only allowed when the 'module' option is set to 'esnext' or 'system', and the 'target' option is set to 'es2017' or higher.ts(1378)`

Same for `processValues`.

What should I do? Sorry, unfortunately I'm not an "async" expert...

Reply:
Hmm yeah good point. That makes this into a way bigger update, as we have to make the environment reading itself async as a whole 🤔 I'll try some things out and report back!

Reply:
Exactly... Thanks! 👍

Reply:
@rijkvanzanten Just let me know if I can support you on this somehow

Reply:
We could maybe get around this by using something like this for now and then switch the environment reading itself to async later on:
```js
env = (async function () {
	await processValues(env);
})();
```

Reply:
I've been thinking about this quite a bit, and there honestly I don't think there's a good solution. You need the environment inside logger to control it's output (raw vs pretty, what level etc), and you need to be able to log messages in the retrieval of the environment. 🥚 🔁 🐔 

By duplicating the logger instance itself, we ignore the configuration completely, which kinda defeats the purpose of using pino for that particular log, as you can no longer accurately store the output using transformers etc (as it's hardcoded to our defaults at that point). 

We can't really make the env reading async either, for similar reasons. We need all environment variables ready to go before anything else will run, in order to ensure that every package and part of code in the tree can access it immediately. 

Like we discussed in [the other PR](https://github.com/directus/directus/pull/6119), the API should stop booting up when a discrepancy is found in the environment variables, as that will cause unexpected / undesirable behavior, so I think the conclusion here is to use regular console logs, and exit the process when appropriate.